### PR TITLE
Sous Chefs Adoption

### PR DESCRIPTION
**Pull request by ramereth**
# Description

Describe what this change achieves

## Issues Resolved

List any existing issues this PR resolves

## Check List

- [ ] All tests pass. See TESTING.md for details.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

### CODE REVIEW

File: attributes/default.rb [hunk 1]
```diff
@@ -17,21 +17,26 @@
...
 
-::Chef::Node.include Opscode::OpenSSL::Password
+# TODO(ramereth): Fix this properly someday
+::Chef::Node.include Opscode::OpenSSL::Password # rubocop:disable ChefCorrectness/OpenSSLPasswordHelpers
 
 if platform_family?('debian')
   default['transmission']['install_method'] = 'package'
   default['transmission']['user']           = 'debian-transmission'
   default['transmission']['group']          = 'debian-transmission'
+elsif platform_family?('rhel', 'amazon')
```
Comment (tas50) [inline attributes/default.rb:27]: should this include fedora as well?

File: libraries/helpers.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+module Transmission
+  module Cookbook
+    module Helpers
+      def transmission_defaults
+        if platform_family?('rhel', 'fedora')
```
Comment (tas50) [inline libraries/helpers.rb:5]: Should these include amazon?

File: resources/torrent_file.rb [hunk 3]
```diff
@@ -24,9 +24,9 @@
...
 attribute :blocking, kind_of: [TrueClass, FalseClass], default: true
-attribute :continue_seeding, default: false
-attribute :owner, regex: Chef::Config[:user_valid_regex]
-attribute :group, regex: Chef::Config[:group_valid_regex]
+attribute :continue_seeding, kind_of: [TrueClass, FalseClass], default: false
+attribute :owner, kind_of: String, regex: Chef::Config[:user_valid_regex]
+attribute :group, kind_of: String, regex: Chef::Config[:group_valid_regex]
 attribute :rpc_host, kind_of: String, default: 'localhost'
 attribute :rpc_port, kind_of: Integer, default: 9091
 attribute :rpc_username, kind_of: String, default: 'transmission'
```
Comment (detjensrobert) [inline resources/torrent_file.rb:32]: Should these attributes get rid of `kind_of` as it isn't needed anymore, or should this wait for a later PR?

Comment (ramereth) [reply resources/torrent_file.rb:32]: IIRC cookstyle fixed this automatically. I'd leave it as is if it's working.

**Review by detjensrobert — APPROVED** [review]
LGTM

File: resources/torrent_file.rb (hunk 3 above)
Comment (tas50) [reply resources/torrent_file.rb:32]: attributes require `kind_of`. Properties use a positional parameter there so the 1st value is the name, the 2nd value is the type, and the next part is a hash of other options.

Comment (detjensrobert) [reply resources/torrent_file.rb:32]: Ah, that makes sense.

**Pull request merged**